src/operations/execute_operation.ts

```diff
@@ -89,19 +89,13 @@ export function executeOperation<
     // that are not explicitly provided with a session.
     let session: ClientSession | undefined = operation.session;
     let owner: symbol | undefined;
-    if (topology.hasSessionSupport()) {
-      if (session == null) {
-        owner = Symbol();
-        session = topology.startSession({ owner, explicit: false });
-      } else if (session.hasEnded) {
-        return cb(new MongoExpiredSessionError('Use of expired sessions is not permitted'));
-      } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
-        return cb(new MongoCompatibilityError('Snapshot reads require MongoDB 5.0 or later'));
-      }
-    } else if (session) {
-      // If the user passed an explicit session and we are still, after server selection,
-      // trying to run against a topology that doesn't support sessions we error out.
-      return cb(new MongoCompatibilityError('Current topology does not support sessions'));
+    if (session == null) {
+      owner = Symbol();
+      session = topology.startSession({ owner, explicit: false });
+    } else if (session.hasEnded) {
+      return cb(new MongoExpiredSessionError('Use of expired sessions is not permitted'));
+    } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
```

Comment:
If any of those mongolike services did give us a `logicalSessionTimeoutMinutes` in the previous code we would have still attached a session so the behaviour in this case would be the same with or without this change.

Reply:
But if they did not give us a `logicalSessionTimeoutMinutes` then we wouldn't have attached one but now we will regardless, usually "extra" data is harmless but it could have implications where folks were relying on sessions being specifically omitted with that case